### ISSUE TITLE
Relaunch Task after cycling the app version in the registry

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud/spring-cloud-dataflow#4240]
Steps to reproduce:

```
1. Register an app
2. Create and launch a new task
3. Destroy the task
4. Remove the app from step 1 and create a new one with the same name but pointing to a new maven url
5.  Create a task using curl command to /tasks/definitions 
6.  While launching the task, customer gets an error saying that the  " Application appname does not exist"
7. Delete all executions from the previous version.
8. Launch the task again and it works after step 7.
```

**Comment by jvalkeal** [comment]
Only time when this can happen is when `pushTaskAppsEnabled` is set to false. It's `true` by default and afaik we never change it.

https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/blob/7faf3264cbc18de5ae7d93d3fba048f397cf7347/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryTaskLauncher.java#L208-L215

Unless `Application appname does not exist` is somewhat thrown directly from cf api and there's a race condition after task is deleted(which would delete app from cf) and new launch request is sent.

Logs would show what actually happens.

**Comment by cppwfs** [comment]
Do they have `spring.cloud.dataflow.task.autocreate-task-definitions` set to true? 

I kicked the tires using just the restful API to see if there was an apparent race condition that could cause the problem above.   Each the dataflow would throw another exception before handing it off to the deployer.   Also from initial review, I can't correlate the creation or deletion of task executions to have anything to do with the launching of tasks.   

Like @jvalkeal said above, logs would be most helpful.

**Comment by jvalkeal** [comment]
Just a comment here as I'm opting to close this as moot as with task overhaul we can now use proper versions, thus all troubles what might come with cycling versions should go away by picking a new version with launch.

**Comment by jvalkeal** [comment]
Closing as we now have a proper support to launch with a different version and there should not be trouble with i.e. old manifest pointing to non-existent version, etc.
